tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -145,20 +145,6 @@ def test_write(self, cloud_storage_type):
             im.evaluate([('vectorized_cloud_storage_index_downloads_total',
                           lambda _, cnt: cnt)]) for im in index_metrics)
 
-        # Matches the segment or the index
-        cache_expr = re.compile(
-            fr'^({self.redpanda.DATA_DIR}/cloud_storage_cache/.*\.log\.\d+)(\.index)?$'
-        )
-
-        # Each segment should have the corresponding index present
-        for node in self.redpanda.nodes:
```

Comment:
It should be possible to keep this validation but make it chunk-aware: the invariant should be that any partitions with >0 chunks should have an index.